src/main/java/io/cryostat/platform/internal/PodmanPlatformClient.java

```diff
@@ -152,99 +167,148 @@ private void queryContainers() {
                     // notifyAsyncTargetDiscovery(EventKind.MODIFIED, sr);
 
                     containers.removeAll(removed);
-                    removed.forEach(
-                            spec -> notifyAsyncTargetDiscovery(EventKind.LOST, convert(spec)));
+                    removed.stream()
+                            .filter(Objects::nonNull)
+                            .forEach(
+                                    spec ->
+                                            notifyAsyncTargetDiscovery(
+                                                    EventKind.LOST, convert(spec)));
 
                     containers.addAll(added);
-                    added.forEach(
-                            spec -> notifyAsyncTargetDiscovery(EventKind.FOUND, convert(spec)));
+                    added.stream()
+                            .filter(Objects::nonNull)
+                            .forEach(
+                                    spec ->
+                                            notifyAsyncTargetDiscovery(
+                                                    EventKind.FOUND, convert(spec)));
                 });
     }
 
-    private void doPodmanRequest(Consumer<List<ContainerSpec>> successHandler) {
+    private void doPodmanListRequest(Consumer<List<ContainerSpec>> successHandler) {
         URI requestPath = URI.create("http://d/v3.0.0/libpod/containers/json");
```

Comment:
I am seeing a new `4.0.0` version for this url: https://docs.podman.io/en/latest/_static/api.html?version=v4.5

Not sure what changes are made but response looks pretty much the same to me. I can also access this with podman `3.4.7`.

Reply:
I've seen that too. I'm not sure what the policy is for podman versions to support older API versions, but I assume that it will remain backward-compatible for some time. So if `3.0.0` does the job and provides the information we need then I think it's better to rely on that version, since more podman versions on more systems will support that than the newer 4.0.0+ API version.

Reply:
Since https://github.com/cryostatio/cryostat/pull/1511 , the platform detection strategy could also be used to check what API version is available at startup, and then pass the minimum supported version to the platform client after it has determined it and checked that it's within a supported range. Something to look into in the future perhaps.

Reply:
Sounds good! Thanks for explaining!